js/web/lib/wasm/jsep/webgpu/ops/slice.ts

```diff
@@ -76,21 +76,38 @@ const fixStartEndValues =
           }
         };
 
+const fillNdArray = (name: string, length: number): string => {
+  let ndArray = `(${name}.x, ${name}.y, ${name}.z, ${name}.w`;
+  for (let i = 0; i < length; i++) {
+    ndArray += `,${name}.${String.fromCharCode(97 + i)}`;
+  }
+  ndArray += ')';
+  return ndArray;
+};
+
 const calculateInputIndicesImpl =
-    (input: IndicesHelper, output: IndicesHelper, inputShape: readonly number[], outputShape: readonly number[],
-     enableInputShapeUniforms: boolean): string =>
-        `fn calculateInputIndices(outputIndices: ${output.type.indices}) -> ${input.type.indices} {
+    (input: IndicesHelper, output: IndicesHelper, inputShape: readonly number[], outputShape: readonly number[]):
+        string => `fn calculateInputIndices(outputIndices: ${output.type.indices}) -> ${input.type.indices} {
           var inputIndices: ${input.type.indices};
           var carry = 0u;
+          ${
+            inputShape.length > 4 ?
+                `let uniforms_input_shape: array<u32, ${inputShape.length}> = array<u32, ${inputShape.length}>${
+                    fillNdArray('uniforms.input_shape', inputShape.length - 4)};
+          let uniforms_steps: array<u32, ${inputShape.length}> = array<u32, ${inputShape.length}>${
+                    fillNdArray('uniforms.steps', inputShape.length - 4)};
+          let uniforms_signs: array<i32, ${inputShape.length}> = array<i32, ${inputShape.length}>${
+                    fillNdArray('uniforms.signs', inputShape.length - 4)};
+          let uniforms_starts: array<u32, ${inputShape.length}> = array<u32, ${inputShape.length}>${
+                    fillNdArray('uniforms.starts', inputShape.length - 4)};` :
+                ''}
+
           for (var i = ${inputShape.length}; i >= 0; i--) {
-            let input_shape_i = ${
```

Comment:
Will it be more simpler if you use it like below?
```
const dimsGet = (index: number, rank: number) : string => {
    if (rank <= 1) return '';
      const vecIndex = ['.x', '.y', '.z', '.w', '.a', '.b'];
      for (let i = 0; i < index - vecIndex.length; i++) {
        vecIndex.push(`.${String.fromCharCode(93 + vecIndex.length + i)}`);
   return vecIndex[i];
}
... ...
const rank = inputShape.length;
... ...
let input_shape_i = ${uniforms.input_shape${dimsGet(i, rank )}};
let steps_i  = ${uniforms.steps${dimsGet(i, rank )}};
let signs_i = ${uniforms.signs${dimsGet(i, rank )}};
...

Reply:
This can not simply be done like this, it has code like this outputIndices[i]. Currently I unrolled the whole loop.